src/main/java/conversion/utils/HttpHelper.java

```diff
@@ -77,32 +73,25 @@ public static int contactCallback(String callbackUrl, Individual individual) thr
     /**
      * This method handles the process of sending the callback data and handles
      * any failed attempts. It will retry once every 10 seconds until it reaches
-     * the MaxRetries, this value is set to 1 by default.
+     * the MaxRetries, this value is set to 3 by default.
      *
      * @param callbackUrl The URL which will receive the json data
-     * @param individual The data to be sent to the callbackUrl
+     * @param jsonData The data to be sent to the callbackUrl
+     * @param ses The executor to add in delays between url calls
+     * @param currentRetries The current count of retries
      */
-    public static void sendCallback(String callbackUrl, Individual individual) {
+    public static void sendCallback(final String callbackUrl, final String jsonData, final ScheduledExecutorService ses, final int currentRetries) {
         try {
-            int resCode = contactCallback(callbackUrl, individual);
-            int currentRetries = 1;
+            int resCode = contactCallback(callbackUrl, jsonData);
 
-            while (resCode != HttpURLConnection.HTTP_OK && !(currentRetries > MaxRetries)) {
+            if (resCode != HttpURLConnection.HTTP_OK) {
                 LOG.log(Level.WARNING, "Callback URL ''{0}'' returned http code: {1} on attempt no.{2}", new Object[]{callbackUrl, Integer.toString(resCode), currentRetries});
 
-                try {
-                    Thread.sleep(10000);
-                } catch (InterruptedException e) {
-                    LOG.info("Callback Retry interupted");
+                if (!(currentRetries >= MaxRetries)) {
```

Comment:
Could this be written as if (currentRetries < MaxRetries)?

Gets rid of the not modifier and is easier to read.

Reply:
Good point, change has been made in latest commit